packages/eslint-plugin/docs/rules/no-empty-object-type.mdx

```diff
@@ -0,0 +1,65 @@
+---
+description: 'Disallow accidentally using the "empty object" type.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-empty-object-type** for documentation.
+
+The `{}`, or "empty object" type in TypeScript is a common source of confusion for developers unfamiliar with TypeScript's structural typing.
+`{}` represents any _non-nullish value_, including literals like `0` and `""`:
+
+```ts
+let anyNonNullishValue: {} = 'Intentionally allowed by TypeScript.';
+```
+
+Often, developers writing `{}` actually mean either:
+
+- `object`: representing any _object_ value
+- `unknown`: representing any value _other than `null` and `undefined`_
+
+To avoid confusion around the `{}` type allowing non-object values, this rule bans usage of the `{}` type.
+
+:::tip
+If you do have a use case for an API allowing any _non-nullish value_, you can always use an [ESLint disable comment](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) or [disable the rule in your ESLint config](https://eslint.org/docs/latest/use/configure/rules#using-configuration-files-1).
+:::
```

Comment:
We don't normally add this tip so prominently to rule docs (it's true for many of them!). But I figured this rule is a special case.

Reply:
Agreed! Though maybe "If you do have a use case for the `{}` type" would be preferable, since we also already suggest `NonNullable<unknown>` as a way to specify "any nonnulish value"? Your call!

Reply:
Hmm yeah I'm on the fence, leaning as-is... we do mention the `{}` in the previous line. I like having it different here both for more pleasant (less repetitive) reading and to enforce that they mean roughly the same thing.